hail/src/main/scala/is/hail/expr/ir/Requiredness.scala

```diff
@@ -638,6 +647,20 @@ class Requiredness(val usesAndDefs: UsesAndDefs, ctx: ExecuteContext) {
         coerce[RStruct](requiredness).field("global").unionFrom(lookup(c).globalType)
       case BlockMatrixToValueApply(child, GetElement(_)) => // BlockMatrix elements are all required
       case BlockMatrixCollect(child) =>  // BlockMatrix elements are all required
+      case ShuffleWith(keyFields, rowType, rowEType, keyEType, name, writer, readers) =>
+        requiredness.union(lookup(writer).required)
+        requiredness.unionFrom(lookup(readers))
```

Comment:
I'm wondering, do we want this level of generality of missingness for writer, id, etc.?  It seems like something is wrong if you ask for the partition bounds with a missing id, for example.  This is much more flexible than the existing code.

Reply:
What happens if one successful partition is missing?